src/embodycodec/types.py

```diff
@@ -248,6 +248,32 @@ class Diagnostics(ComplexType):
     avg_voltage: int
 
 
+@dataclass
+class BatteryDiagnostics(ComplexType):
+    struct_format = "<IIHHhhHHHH"
```

Comment:
Hvorfor den andre dekodingen?

Reply:
Det er separate implementasjoner for fil-codec og wire. Opprinnelig var årsaken at det blant annet var 3 byte ppg-verdier i file-codec som ikke støttes av struct sin pack/unpack. Men inntil jeg tar jobben med å refaktorere er det forskjellige implementasjoner.

Reply:
Men hadde jeg brukt en halvtime ekstra kunne jeg brukt struct unpack for denne meldingstypen i stedet for å parse manuelt i file-codecen ja.